src/algorithms/rsa.rs

```diff
@@ -194,3 +195,87 @@ fn blind<R: CryptoRngCore, K: PublicKeyParts>(
 fn unblind(key: &impl PublicKeyParts, m: &BigUint, unblinder: &BigUint) -> BigUint {
     (m * unblinder) % key.n()
 }
+
+/// The following (deterministic) algorithm also recovers the prime factors `p` and `q` of a modulus `n`, given the
+/// public exponent `e` and private exponent `d`.
+pub fn recover_primes(n: &BigUint, e: &BigUint, d: &BigUint) -> Result<(BigUint, BigUint)> {
+    // 1. Let a = (de – 1) × GCD(n – 1, de – 1).
+    let mut a = (d * e - BigUint::one()) * (n - BigUint::one()).gcd(&(d * e - BigUint::one()));
+
+    // 2. Let m = floor(a /n) and r = a – m n, so that a = m n + r and 0 ≤ r < n.
+    let mut m = &a / n;
+    let mut r = &a - &m * n;
+    a.zeroize();
```

Comment:
[`Zeroizing`](https://docs.rs/zeroize/latest/zeroize/struct.Zeroizing.html) will allow you to eliminate the manual `zeroize` calls:

```suggestion
    let mut a = Zeroizing::new(d * e - BigUint::one()) * (n - BigUint::one()).gcd(&(d * e - BigUint::one())));

    // 2. Let m = floor(a /n) and r = a – m n, so that a = m n + r and 0 ≤ r < n.
    let mut m = &a / n;
    let mut r = &a - &m * n;
```

Reply:
I think this should do it:
https://github.com/RustCrypto/RSA/pull/380/commits/7fe9a5747c6a735346e2f2a08a1174829bf443a8